standard/ERCs/erc-6900.md

```diff
@@ -537,36 +526,51 @@ Finally, the function MUST emit the event `ModuleUninstalled` with the module's
 
-The set of hooks run for a given execution function MUST be the hooks specified by account state at the start of the execution phase. This is relevant for functions like `installModule` and `uninstallModule`, which modify the account state, and possibly other execution or native functions as well.
+Prior to running the target function, the modular account MUST run all pre execution hooks that apply for the current function call. Pre execution hooks apply if they have been installed to the currently running function selector, or if they are installed to the validation function that was used for the current execution. Pre execution hooks MUST run validation-associated hooks first, in the order they were installed, then selector-associated hooks second, in the order they were installed.
 
-#### Calls made from modules
+Next, the modular account MUST run the target function, either an account native function or a module-defined execution function.
 
-Modules MAY interact with other modules and external addresses through the modular account using the functions defined in the `IModuleExecutor` interface. These functions MAY be called without a defined validation function, but the modular account MUST enforce these checks and behaviors:
+After the execution of the target function, the modular account MUST run any post execution hooks. These MUST be run in the reverse order of the pre execution hooks. If a hook is defined to be both a pre and a post execution hook, and the pre execution hook returned a non-empty `bytes` value to the account, the account MUST pass that data to the post execution hook.
 
-The `executeFromModule` function MUST allow modules to call execution functions installed by modules on the modular account. Hooks matching the function selector provided in `data` MUST be called. If the calling module's manifest did not include the provided function selector within `permittedExecutionSelectors` at the time of installation, execution MUST revert.
+The set of hooks run for a given target function MUST be the hooks specified by account state at the start of the execution phase. In other words, if the set of applicable hooks changes during execution, the original set of hooks MUST still run, and only future invocations of the same target function should reflect the changed set of hooks.
 
-The `executeFromModuleExternal` function MUST allow modules to call external addresses as specified by its parameters on behalf of the modular account. If the calling module's manifest did not explicitly allow the external call within `permittedExternalCalls` at the time of installation, execution MUST revert.
+Module execution functions where the field `isPublic` is set to true, or native functions without access control, SHOULD omit the runtime validation step, including any runtime validation hooks. Native functions without access control MAY also omit running execution hooks.
```

Comment:
We should state the risk where if the `isPublic` function changes states, omitting validation can be dangerous.

Reply:
Do we need that extra detail? I think that is obvious from the spec stating that validation can be skipped if the function is marked public.

Reply:
The code comment on `isPublic` is `Whether or not the function needs runtime validation, or can be called by anyone. `. 
Why didn't we name it `skipRuntimeValidation` flag?

My concern of the name `isPublic` is that it does not indicate that it can potentially be state changing. We only skip runtime by using the old `ALWAYS_ALLOW`.

Reply:
> Why didn't we name it `skipRuntimeValidation` flag?

That does seem like a good idea, makes it more literal. Maybe less obvious to connect it to view functions, but it would match the naming of other flags.

Reply:
>Maybe less obvious to connect it to view functions

They don't have to be view functions is the whole point (to allow state changing though I am not sure what the use cases are, maybe exec timelock tx after trigger time or sth), right?
Otherwise, it can just be `isPublic` or `isPublicView` or `isView`, etc.

Reply:
Makes sense. Let's update that in a different PR, because it also requires code changes.